backend/src/services/tasks.service.ts

```diff
@@ -1,16 +1,26 @@
 import { Injectable } from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
+import { Comment } from 'src/entities/comment.entity';
 import { Task } from 'src/entities/task.entity';
 import { Repository } from 'typeorm';
 
 @Injectable()
 export class TasksService {
   constructor(
-    @InjectRepository(Task) private readonly tasksRepository: Repository<Task>,
+    @InjectRepository(Task)
+    public readonly tasksRepository: Repository<Task>,
+    @InjectRepository(Comment)
+    public readonly commentsRepository: Repository<Comment>,
   ) {}
 
-  getTasks(): Promise<Task[]> {
-    return this.tasksRepository.find();
+  async getTasks(): Promise<Task[]> {
+    const tasks = await this.tasksRepository.find();
+    const updateTasksJob = tasks.map(async (task) => {
+      const comments = await this.commentsRepository.find({ where: { task } });
+      task.comments = comments;
+    });
+    await Promise.allSettled(updateTasksJob);
+    return tasks;
```

Comment:
Although it works, it may not be the best approach with regard to performance as it will load all tasks and then, for each task, open a new connection to the DB, load its comments and close the connection.
Ideally we should load the tasks as well as their comments in just one shot doing something like this:
```
const comments = await this.commentsRepository.find({ where: { task }, relations: ["comments"] });
```